index.js

```diff
@@ -11,11 +13,36 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
-
+function validatePaymentDataMiddleWare(req, res, next) {
+    const payment = {...req.body};
+    const currentDate = new Date().toLocaleDateString('fr-CA');
+    
+    if(!(payment.date >= currentDate &&
+        currencies.includes(payment.currency) &&
+        Number.isFinite(payment.amount) && 
+        Number.isFinite(payment.exchangeRate) &&
+        payment.description &&
+        payment.description !== '')) {
+            res.status(400).send('Invalid Data');
+        };
+    next();
```

Comment:
`next()` gets some param. please try debugging it and find out what is that param.
also, notice what happens if description is wrong and `next()`  is being called with no params.
implementation of method `next()` we can find in file: `node_modules\express\lib\router\route.js`